components/locid/src/extensions/mod.rs

```diff
@@ -164,9 +185,25 @@ impl core::fmt::Display for Extensions {
 
 impl writeable::Writeable for Extensions {
     fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
-        // Alphabetic by singleton (t, u, x)
-        writeable::Writeable::write_to(&self.transform, sink)?;
-        writeable::Writeable::write_to(&self.unicode, sink)?;
+        let mut wrote_tu = false;
+        self.other.iter().try_for_each(|other| {
+            if other.get_ext() > 't' && !wrote_tu {
+                writeable::Writeable::write_to(&self.transform, sink)?;
+                writeable::Writeable::write_to(&self.unicode, sink)?;
```

Comment:
Suggestion (optional): It is convenient that `t` and `u` are next to each other in the alphabet.  If they were separate, I think you would need more complicated logic here.  Consider adding a comment explaining that?